src/sqlfluff/rules/L028.py

```diff
@@ -58,68 +78,171 @@ def _lint_references_and_aliases(
-                            "inconsistent with previous references.",
-                        )
-                    )
-            elif self.single_table_references != this_ref_type:
-                violation_buff.append(
-                    LintResult(
-                        anchor=ref,
-                        description="{} reference {!r} found in single table "
-                        "select.".format(this_ref_type.capitalize(), ref.raw),
-                    )
-                )
-            seen_ref_types.add(this_ref_type)
-
-        return violation_buff or None
+        self.single_table_references: str
+        # This could be turned into an option
+        fix_inconsistent_to = "qualified"
+        return _generate_fixes(
+            table_aliases,
+            standalone_aliases,
+            references,
+            col_aliases,
+            self.single_table_references,
+            self._allow_lateral_reference,
+            self._is_struct_dialect,
+            fix_inconsistent_to,
+        )
 
     def _eval(self, context: RuleContext) -> EvalResultType:
         """Override base class for dialects that use structs, or SELECT aliases."""
         # Config type hints
         self.force_enable: bool
-
+        self.dialect: str = context.dialect.name
         # Some dialects use structs (e.g. column.field) which look like
         # table references and so incorrectly trigger this rule.
         if (
-            context.dialect.name in ["bigquery", "hive", "redshift"]
+            context.dialect.name in self._dialects_with_structs
             and not self.force_enable
         ):
             return LintResult()
 
-        # Certain dialects allow use of SELECT alias in WHERE clauses
-        if context.dialect.name in ["snowflake", "redshift"]:
-            self._allow_select_alias = True
+        # See comment above (by prop definition)
+        if context.dialect.name in self._dialects_allowing_lateral_reference:
+            self._allow_lateral_reference = True
+
+        if context.dialect.name in self._dialects_with_structs:
+            self._is_struct_dialect = True
 
         return super()._eval(context=context)
+
+
+def _generate_fixes(
+    table_aliases: List[AliasInfo],
+    standalone_aliases: List[str],
+    references: List[BaseSegment],
+    col_aliases: List[ColumnAliasInfo],
+    single_table_references: str,
+    allow_select_alias: bool,
+    is_struct_dialect: bool,
+    fix_inconsistent_to: Optional[str],
+) -> Optional[List[LintResult]]:
+    """Iterate through references and check consistency."""
+    # How many aliases are there? If more than one then abort.
+    if len(table_aliases) > 1:
+        return None
+    # A buffer to keep any violations.
+    violation_buff: List[LintResult] = []
+    col_alias_names: List[str] = [c.alias_identifier_name for c in col_aliases]
+    table_ref_str: str = table_aliases[0].ref_str
```

Comment:
Is it possible that `table_aliases` has length 0? If so, this would crash with an `IndexError`.

Reply:
Not sure but will guard